virttest/test_setup.py

```diff
@@ -434,6 +434,12 @@ def __init__(self, params):
         vm_mem_minimum = params.get("vm_mem_minimum", "512M")
         self.lowest_mem_per_vm = float(normalize_data_size(vm_mem_minimum))
         utils_memory.drop_caches()
+        hugepages_surp = int(params.get("hugepages_surp", 0))
+        if hugepages_surp != self.get_hugepages_surp():
+            with open("/proc/sys/vm/nr_overcommit_hugepages", 'r+') as over_commit:
```

Comment:
I thought of something like this,

**Plain and straight forward**
```python
self.hugepages_surp = int(params.get("hugepages_surp", 0))
if self.hugepage_surp != int(utils_memory.read_from_meminfo('HugePages_Surp')):
    logging.warn(...)
    over_commit = kernel_interface.ProcFS("/proc/sys/vm/nr_overcommit_hugepages")
    over_commit.proc_fs_value = self.hugepages_surp
```

**To consider testing hugepages_surp feature,**
```python
self.hugepages_surp = int(params.get("hugepages_surp", 0))
if self.hugepage_surp != int(utils_memory.read_from_meminfo('HugePages_Surp')):
    logging.warn(...)
    over_commit = kernel_interface.ProcFS("/proc/sys/vm/nr_overcommit_hugepages")
    over_commit.proc_fs_value = self.hugepages_surp
:::
:::
# By default `hugepages_surp` would be 0, but if we need to test it
# set the param `hugepages_surp` so that we can take target hugepages
# subtracting hugepages_surp and kernel is expected to allocate rest of
# them from normal pages pool
self.target_hugepages = target_hugepages - self.hugepages_surp
```

Reply:
After VM boots, we should be able to see `Hugepages_Surp` in /proc/meminfo being updated by kernel with `self.hugepages_surp`

Reply:
I don't think so. If we want to get `Hugepages_Surp` after guest boots, we can invoke `get_hugepages_surp()` in the test case